### PR TITLE
Обновление парсера

### PR DESCRIPTION
**Pull request by theshadowco**
## Описание

Доработки, связанные с обновлением bsl parser
- изменения в лексике SDBL парсера
- исправление некоторых ошибок
- новый ANTLR + null check + фиксы из-за нарушения обратной совместимости

## Связанные задачи

Closes 

### Общие

- [x] Ветка PR обновлена из develop
- [x] Отладочные, закомментированные и прочие, не имеющие смысла участки кода удалены
- [x] Изменения покрыты тестами
- [ ] Обязательные действия перед коммитом выполнены (запускал команду `gradlew precommit`)

### Для диагностик

- [ ] Описание диагностики заполнено для обоих языков (присутствуют файлы для обоих языков, для русского заполнено все подробно, перевод на английский можно опустить)

## Summary by CodeRabbit

* **New Features**
  * Broadened SDBL tokenization and keyword/join recognition for richer language support.

* **Bug Fixes**
  * Improved null-safety across parsing, diagnostics and highlighting to reduce errors.
  * Tightened control-flow graph and query diagnostics to avoid edge-case failures.

* **Chores**
  * Upgraded parser dependency and added support configuration library.

* **Tests**
  * Updated token/highlight tests to reflect merged multi-token keyword handling.

### CODE REVIEW

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/aop/sentry/PermissionFilterBeforeSendCallback.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ private CompletableFuture<MessageActionItem> askUserForPermission(LanguageClient
     return languageClient.showMessageRequest(requestParams);
   }
 
-  @Nullable private MessageActionItem waitForPermission(CompletableFuture<MessageActionItem> sendQuestion) {
+  private @Nullable MessageActionItem waitForPermission(CompletableFuture<@Nullable MessageActionItem> sendQuestion) {
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/aop/sentry/PermissionFilterBeforeSendCallback.java]: Откуда @nullable в параметре?

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/DoubleNegativesDiagnostic.java [hunk 2]
```diff
@@ -86,7 +86,12 @@ private static boolean isNegationOperator(BslExpression parent) {
   }
 
   private void addDiagnostic(BinaryOperationNode node) {
-    var startToken = Trees.getTokens(node.getParent().getRepresentingAst())
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/DoubleNegativesDiagnostic.java:89]: @theshadowco тут node - это кусок cfg, а не parse tree. Оно точно @nullable?

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/IfElseDuplicatedCodeBlockDiagnostic.java [hunk 3]
```diff
@@ -61,7 +62,7 @@ public void init() {
   }
 
   @Override
-  public ParseTree visitIfStatement(BSLParser.IfStatementContext ctx) {
+  public @Nullable ParseTree visitIfStatement(BSLParser.IfStatementContext ctx) {
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/IfElseDuplicatedCodeBlockDiagnostic.java:65]: @theshadowco почему visit-методы стали @nullable для возвращаемого значения? Они же вызываются только, когда ctx не равен null, значит, должны вызвать либо super, либо вернуть себя же, то есть не-null.

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/aop/sentry/PermissionFilterBeforeSendCallback.java (hunk 1 above)
Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/aop/sentry/PermissionFilterBeforeSendCallback.java]: хороший вопрос... не помню, уберу

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/DoubleNegativesDiagnostic.java (hunk 2 above)
Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/DoubleNegativesDiagnostic.java:89]: ну parent может быть null... это например рут

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/IfElseDuplicatedCodeBlockDiagnostic.java (hunk 3 above)
Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/IfElseDuplicatedCodeBlockDiagnostic.java:65]: забываешь про рут.
Там схема вызова построена как раз на наличии null в результате - если он есть, то значит дальше нет никто.

Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/IfElseDuplicatedCodeBlockDiagnostic.java:65]: ну и да, прервать дальнейшее движение по дереву можно именно с null (кейсов не придумаю, но так есть)

Comment (theshadowco) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/IfElseDuplicatedCodeBlockDiagnostic.java:65]: попробую поковырять antlr
